lightning-invoice/src/utils.rs

```diff
@@ -1015,4 +1201,81 @@ mod test {
 		}
 		assert!(chan_ids_to_match.is_empty(), "Unmatched short channel ids: {:?}", chan_ids_to_match);
 	}
+
+    #[test]
+    fn test_create_invoice_with_description_hash() {
+		let chanmon_cfgs = create_chanmon_cfgs(2);
+		let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+		let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+		let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+		let invoice = ::utils::create_invoice_from_channelmanager_with_description_hash_and_duration_since_epoch(
+			&nodes[1].node, nodes[1].keys_manager, Currency::BitcoinTestnet, Some(10_000),
+			"Testing description_hash".to_string(), Duration::from_secs(1234567),
+        )
+        .unwrap();
+        assert_eq!(invoice.amount_pico_btc(), Some(100_000));
+        assert_eq!(
+            invoice.min_final_cltv_expiry(),
+            MIN_FINAL_CLTV_EXPIRY as u64
+        );
+        assert_eq!(
+            invoice.description(),
+            InvoiceDescription::Hash(&crate::Sha256(Sha256::hash(
+                "Testing description_hash".as_bytes()
+            )))
+        );
+	}
+
+
+	#[test]
+    #[cfg(feature = "std")]
+    fn create_phantom_invoice_with_description_hash() {
```

Comment:
Same here, I suggest that you place this right after the `do_test_multi_node_receive` test.

Reply:
[d1f4536](https://github.com/lightningdevkit/rust-lightning/pull/1361/commits/d1f4536ddecfde661b99777903a9af14b58064f7)